integreat_cms/cms/views/media/media_actions.py

```diff
@@ -275,6 +274,20 @@ def replace_file_ajax(request, region_slug=None):
             status=400,
         )
 
+    if not request.user.has_perm("cms.replace_mediafile"):
+        return JsonResponse(
+            {
+                "messages": [
+                    {
+                        "type": "error",
+                        "text": _(
+                            "You don't have the permission to replace media files."
+                        ),
+                    }
+                ],
+            }
+        )
```

Comment:
If I got it right, this error message is never shown to the user anyway, because the replace button isn't shown for users who don't have the permission to replace media files, right? Then I guess we don't need the verbose error message and a generic 403 error raised by the more concise `@permission_required` decorator is sufficient...

Reply:
hmm, somehow the 403 page will not be called as I tried with @permission_required, showing the button temporary for the observer user and using it to emulate the case an observer makes it to send an delete request 🤔

Reply:
I'm not sure what you mean, if the `@permission_required` really doesn't work, we have a very big problem :sweat_smile:
We also use this decorator in a lot of other views, would be a very critical bug...
Maybe it was connected to the mistake in the migration, so that the observer user still had the permission to replace media files?

Reply:
It would be very cool to have view tests for the media actions to be absolutely sure we're not messing something up here, but that's not completely trivial... I guess I'll open another issue for it :grin:

Reply:
For reference: I opened #1997 for that matter.

Reply:
Thank you for opening an issue for that 👍 

It seems @permission_required itsself is indeed working. I got the 403 error page as I tried to publish a page as observer without permission for it.

However the 403 error page will not be shown when a delete file request is sent by an observer. This remains so also after modification of the migration. 
With @permission_required it returns 403 forbidden to file deleting requests, though no redirect to the 403 error page. I think it has something to do with preact, for the console shows "Uncaught (in promise) Error: 403: Forbidden" but I don't know how and what to change 😓

Reply:
Ah ok, so the user is correctly prevented from replacing/deleting files (even if they manage to show the button in the developer tools), this is all we're caring about.
It doesn't have to look nice in this case, because it will only happen if the user tampers with the HTML source code. Users who do that are not entitled to nice error messages :sweat_smile:

Reply:
Yes, permission check and resulting rejection of permissionless delete/replace requests are successfull.
I'll implement with @permission_required. It brings more consistency 🙂